common-compose.yml

```diff
@@ -0,0 +1,53 @@
+# Elasticsearch stack designed for container-native deployment
+# on Joyent's Triton platform.
+
+# ---------------------------------------------------
+# This common service acts as a template for master-only, data-only, and
+# master+data ES services defined in the docker-compose.yml file.
+
+elasticsearch:
+    image: 0x74696d/triton-elasticsearch:latest
+    mem_limit: 4g
+    restart: always
+
+    environment:
+    - CLUSTER_NAME=elasticsearch
+    - ES_HEAP_SIZE=1g # set to 50% of mem_limit, but no more than 31g
+    - ES_NODE_MASTER=true # default
+    - ES_NODE_DATA=true   # default
+    - CONTAINERBUDDY=file:///etc/containerbuddy.json
+    # ES_HEAP_NEWSIZE # optional override
+    # ES_DIRECT_SIZE  # optional override
+    # ES_JAVA_OPTS    # optional override
```

Comment:
We will need to set the following JAVA_OPTS in order for ES to behave properly:

N is typically calculated as follows:

Default Value:
ParallelGCThreads = (ncpus <= 8) ? ncpus : 3 + ((ncpus * 5) / 8)

Defaults to 30 on a SmartOS box. Use the formula above to calculate for the amount of vCPUs

Default Value:
ParallelCMSThreads = (ParallelGCThreads + 3) / 4

You can use a script like: https://github.com/joyent/solutions-engineering/blob/master/toolbox/bin/proclimit to build an approximate base value that you then compute from.

```
-XX:-UseGCTaskAffinity -XX:-BindGCTaskThreadsToCPUs -XX:ParallelGCThreads=N
```